src/app/clusters/ota-requestor/OTARequestor.cpp

```diff
@@ -410,3 +380,37 @@ void OTARequestor::TriggerImmediateQuery()
     // Perhaps we don't need a separate function ConnectToProvider, revisit this
     ConnectToProvider();
 }
+
+CHIP_ERROR OTARequestor::BuildQueryImageRequest(QueryImageRequest & request)
+{
+    constexpr EmberAfOTADownloadProtocol kProtocolsSupported[] = { EMBER_ZCL_OTA_DOWNLOAD_PROTOCOL_BDX_SYNCHRONOUS };
+    constexpr bool kRequestorCanConsent                        = false;
+    QueryImage::Type & args                                    = request.args;
+
+    uint16_t vendorId;
+    VerifyOrReturnError(Basic::Attributes::VendorID::Get(kRootEndpointId, &vendorId) == EMBER_ZCL_STATUS_SUCCESS,
+                        CHIP_ERROR_INTERNAL);
```

Comment:
nit: internal error seems very vague. Any chance we can return something that implies  more of a "cannot get mandatory piece of data", this may be even a new CHIP_ERROR like `CHIP_ERROR_REQUIRED_DATA_NOT_AVAILABLE` or similar.

Reply:
Found CHIP_ERROR_READ_FAILED